Error when applying castNonNull() to a @MonotonicNonNull type

E.g.

```
import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
import org.checkerframework.checker.nullness.NullnessUtil;

class Test {

    @MonotonicNonNull
    private String x;

    String test() {
        return NullnessUtil.castNonNull(x);
    }
}
```

Results in the following error:

```
Test.java:10: error: [monotonic.type.incompatible] cannot assign org.checkerframework.checker.nullness.qual.MonotonicNonNull to org.checkerframework.checker.nullness.qual.MonotonicNonNull (monotonic type).
        return NullnessUtil.castNonNull(x);
                                        ^
  full type found: @Initialized @MonotonicNonNull String
1 error
```

--- a/checker/src/org/checkerframework/checker/nullness/NullnessUtil.java
+++ b/checker/src/org/checkerframework/checker/nullness/NullnessUtil.java
@@ -71,7 +71,7 @@ public final class NullnessUtil {
      * @return the argument, casted to have the type qualifier @NonNull
      */
     public static @EnsuresNonNull("#1") <T extends /*@Nullable*/ Object> /*@NonNull*/ T castNonNull(
-            T ref) {
+            /*@Nullable*/ T ref) {
         assert ref != null : "Misuse of castNonNull: called with a null argument";
         return (/*@NonNull*/ T) ref;
     }

--- a/checker/tests/nullness/Issue1555.java
+++ b/checker/tests/nullness/Issue1555.java
@@ -1,0 +1,11 @@
+import org.checkerframework.checker.nullness.NullnessUtil;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+class Issue1555 {
+
+    @MonotonicNonNull private String x;
+
+    String test() {
+        return NullnessUtil.castNonNull(x);
+    }
+}

--- a/checker/tests/nullness/Issue1555.java
+++ b/checker/tests/nullness/Issue1555.java
@@ -1,9 +1,11 @@
+// Test case for Issue 1555
+// https://github.com/typetools/checker-framework/issues/1555
 import org.checkerframework.checker.nullness.NullnessUtil;
 import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
 
 class Issue1555 {
 
-    @MonotonicNonNull private String x;
+    private @MonotonicNonNull String x;
 
     String test() {
         return NullnessUtil.castNonNull(x);

